src/app/lib/business/environment/environment_manager.dart

```diff
@@ -0,0 +1,70 @@
+import 'package:app/access/environment/environment_repository.dart';
+import 'package:flutter_dotenv/flutter_dotenv.dart';
+
+/// Service to use as the source of truth for application environment (e.g. Staging, Production, etc.).
+abstract interface class EnvironmentManager {
+  factory EnvironmentManager(EnvironmentRepository repository, DotEnv dotEnv) =
+      _EnvironmentManager;
+
+  /// Gets the current environment (which can be overridden).
+  String get current;
```

Comment:
Wondering if we could use an `enum` instead of a string. It could look like the following.

``` dart
/// Environments that the application can be run in.
enum Environment {
  development,
  qa,
  staging,
  production;

  @override
  String toString() => switch (this) {
        development => 'Development',
        staging => 'Staging',
        production => 'Production',
      };
}
```

Reply:
I need to document it now

Reply:
Done!